# STATISTICS AND PROBABILITY

 **1.The probability that one card is diamond,one card is heart and one card is spade**


The probability of choosing the diamond,P(diamond)=13/52=0.25

without replacement you now have 51 cards left
so the probability of choosing the heart,P(heart)=13/51=0.2549

without replacement you now have 50 cards left
so the probability of choosing the spade,P(spade)=13/50=0.26

The probability that one card is diamond,one card is heart and one card is spade=P(diamond)*P(heart)*P(spade)=0.25*0.2549*2.26 = 0.0165685

**1 ANSWER: 
The probability that one card is diamond,one card is heart and one card is spade = 0.0165685**




**2.a)Favorite movie type is either action or drama movie**

The probability of action movie =42/100





The probability of drama movie =36/100

The probability of either action or drama=(42/100)+(36/100)=78/100 = 0.78

**b)Favorite movie type is either comedy or horror movie**

The probability of comedy movie =54/100

The probability of horror movie =12/100

The probability of either commedy or horror=(54/100)+(12/100)=66/100 = 0.66

**2  ANSWER:
a)Favorite movie type is either action or drama movie = 0.78
b)Favorite movie type is either comedy or horror movie = 0.66**

**3.A drawn ball is to be observed as black,the probability of bag B to be selected**

**A bag**
A bag contains red balls=3,
A bag contains black balls=5

**B bag**
B bag contains white balls=4,
B bag contains black balls=7


The probability of black balls=P(A)=P(B)=1/2

The probablity of black balls in A bag = P(B/A) = 5/8

The probablity of black balls in B bag = P(B/B) = 7/11

The probability of drawn ball from B = P(D/B) = ( P(B) * P(B/B) ) / ( P(A) * P(B/A) ) + ( P(B) * P(B/B) )

 = ( (1/2) * (7/11) ) / ( 1/2 * 5/8 ) + ( 1/2 * 7/11 )
                                              =  0.5045

**3 ANSWER:The probability of drawn ball from B = 0.5045**

**4. a)Receivinng 10 application in 1 minute**

 λ=450/6
        λ=15/2, x=10
                 
                 e**(-15/2)(15/2)**10
        P(X=x) = _________________________
                          10!
        P(X=10) = 0.0858

**b) Receiving atleat 17 applications in 2 minutes**

x=17
                 

> 




                  e**(-15/2)(15/2)**17
        P(X=x) = _________________________
                          17!
        P(X=17)= 0.6321

**4 ANSWER: a)Receivinng 10 application in 1 minute = 0.0858,  b)Receiving atleat 17 applications in 2 minutes = 0.6321**

**6.75th percentile of prices among the new home in Manchester**

z=0.675
μ=350870	
σ=12405
z=(x-μ)/σ


0.675=(x-350870)/12405


x=350870+(0.675*12405)


x=359237.0456

**6 ANSWER:75th percentile of prices among the new home in Manchester=359237.0456**

**7. Commpute the value of F-Statistics using the relation F=t2**

In [1]:
import numpy as np
import scipy.stats
union_city = [44,21,36,34,77,32,29,39,92,67,45,37]
mountain_view = [47,81,41,79,38,20,33,83,41,40,69,36]

def f_test(x, y):
    x = np.array(x)
    y = np.array(y)
    f = np.var(x, ddof=1)/np.var(y, ddof=1)
    dfn = x.size-1 
    dfd = y.size-1 
    p = 1-scipy.stats.f.cdf(f, dfn, dfd) 
    return f, p
f,p = f_test(union_city, mountain_view)
print('F-statistics:',round(f,5))

F-statistics: 0.99212


**7 ANSWER:F-statistics: 0.99212**

**8. Calculate F-statistics for groups**



In [2]:
import numpy as np
import scipy.stats
# group 1, group 2, group 3
mean = [44.5,42,46.5]
std_dev = [5,5,8]
# Sample size is same for all three groups so leave it...
def f_test(x, y):
    x = np.array(x)
    y = np.array(y)
    f = np.var(x, ddof=1)/np.var(y, ddof=1)
    dfn = x.size-1 
    dfd = y.size-1 
    p = 1-scipy.stats.f.cdf(f, dfn, dfd) 
    return f, p

f,p = f_test(mean, std_dev)

print('F-statistics:',round(f,5))

F-statistics: 1.69444


**8 ANSWER:F-statistics - 1.69444**

**9.The probability that the number of pizza exceeds 80 in a day**

In [3]:
from scipy.stats import poisson
poisson_prob=poisson.pmf(k=80,mu=76)
print(round(poisson_prob,5))

0.04018


**9 ANSWER:The probability that the number of pizza exceeds 80 in a day:0.04018**

**10.Protein suplliers affect diet or not?**

In [4]:
before = [82 for j in range(25)]
after = [95 for i in range(25)]
from scipy.stats import ttest_ind

if ttest_ind(before,after)[-1] <= 0.05 :
    print('protein supplier affect diet')
else:
    print('protein supplier did not affect diet')

protein supplier affect diet


**10 ANSWER:Protein supplier affect diet**

**11. Test the claim that the different days of the week have same frequency of the customers**

In [5]:
no_of_customers = [1419,1320,1526,1289,1620,2067,2055]
avg = sum(no_of_customers)/len(no_of_customers)
from scipy.stats import ttest_1samp
test = ttest_1samp(no_of_customers,avg)
print(test)
if test[-1] <= 0.1 :
    print('Different days of the week have the same frequency of customers')
else:
     print('Different days of the week have the different frequency of customers')

Ttest_1sampResult(statistic=0.0, pvalue=1.0)
Different days of the week have the different frequency of customers


**11 ANSWER:Different days of the week have the different frequency of customers**

**12. Is death distribution of current months is same as expert's  prediction**

In [6]:
import pandas as pd
City = ["New York", "New Jersey", "Michigan", "California", "Florida", "Massachusetts", "Texas"]
No_of_deaths_in_last_month = [3406, 1469, 662, 583, 582, 526, 461]
No_of_deaths_in_current_month = [4398, 1846, 1288, 382, 879, 430, 321]
df = pd.DataFrame({"City": City, "No_of_deaths_in_last_month":No_of_deaths_in_last_month, "No_of_deaths_in_current_month": No_of_deaths_in_current_month})
print(df)
print()

predicted=[]
significance=[]
for i in No_of_deaths_in_last_month:
    predicted.append((0.30*i)+i)
    significance.append(round(0.1*i,1))
print("predicted by scientist=",predicted)
print("actual current death value=",No_of_deaths_in_current_month)

import numpy as np
n_predicted=np.array(predicted)
n_No_of_deaths_in_current_month=np.array(No_of_deaths_in_current_month)
difference=np.absolute(n_No_of_deaths_in_current_month-n_predicted)
print("difference between predicted and current=",difference)
print("signicance of 0.1=",significance)
print()

from scipy.stats import ttest_ind
print(ttest_ind(No_of_deaths_in_current_month,predicted))
print()
if (ttest_ind(No_of_deaths_in_current_month,predicted)[-1] <=0.1 ):
    print("Expert's prediction is same")
else:
    print("Expert's prediction is not same")

            City  No_of_deaths_in_last_month  No_of_deaths_in_current_month
0       New York                        3406                           4398
1     New Jersey                        1469                           1846
2       Michigan                         662                           1288
3     California                         583                            382
4        Florida                         582                            879
5  Massachusetts                         526                            430
6          Texas                         461                            321

predicted by scientist= [4427.8, 1909.7, 860.6, 757.9, 756.6, 683.8, 599.3]
actual current death value= [4398, 1846, 1288, 382, 879, 430, 321]
difference between predicted and current= [ 29.8  63.7 427.4 375.9 122.4 253.8 278.3]
signicance of 0.1= [340.6, 146.9, 66.2, 58.3, 58.2, 52.6, 46.1]

Ttest_indResult(statistic=-0.08484138410174252, pvalue=0.933786528927488)

Expert's prediction is

**12 ANSWER:Expert's prediction is not same**